src/source/source_state.js

```diff
@@ -62,12 +62,12 @@ class SourceFeatureState {
 
         this.deletedStates[sourceLayer] = this.deletedStates[sourceLayer] || {};
 
-        if (key && featureId) {
+        if (key && typeof featureId === 'number' && featureId >= 0) {
```

Comment:
Line 61 already converts the `featureId` function parameter to a string. It should be enough to switch from `featureId` to `feature` in these if clauses.

Reply:
`typeof feature` will be `string` then. Generally, for a targeted fix of `removeFeatureState` not working with `0` id, it would be enough to do `featureId !== undefined` for consistency (we don't check validity in `setFeatureState` either).

Reply:
>it would be enough to do featureId !== undefined

was originally gonna say that wouldn't catch id's like boolean types, but both implementations would treat booleans as invalid id's and therefore interpret the deletion as source-wide deletion 👍